Counterexamples/DiscreteTopologyWithNontrivialValuation.lean

```diff
@@ -0,0 +1,109 @@
+import Mathlib.FieldTheory.RatFunc.AsPolynomial
+import Mathlib.RingTheory.Valuation.RankOne
+import Mathlib.Topology.Algebra.Valued.ValuativeRel
+
+-- MOVE
+namespace Valuation
+
+variable {R Γ₀ Γ'₀ : Type*} [Ring R]
+  [LinearOrderedCommMonoidWithZero Γ₀] [LinearOrderedCommMonoidWithZero Γ'₀]
+  {v₁ : Valuation R Γ₀} {v₂ : Valuation R Γ'₀}
+
+attribute [mk_iff] Valuation.IsNontrivial
+
+lemma IsEquiv.isNontrivial_iff_isNontrivial (hv : v₁.IsEquiv v₂) :
+    v₁.IsNontrivial ↔ v₂.IsNontrivial := by
+  rw [isNontrivial_iff, isNontrivial_iff]
+  simp_rw [ne_eq, ← v₁.map_zero, ← v₁.map_one, ← v₂.map_zero, ← v₂.map_one, hv.val_eq]
+
+lemma IsEquiv.isNontrivial (hv : v₁.IsEquiv v₂) [v₁.IsNontrivial] : v₂.IsNontrivial :=
+  hv.isNontrivial_iff_isNontrivial.mp ‹_›
+
+lemma IsEquiv.isNontrivial' (hv : v₁.IsEquiv v₂) [v₂.IsNontrivial] : v₁.IsNontrivial :=
+  hv.isNontrivial_iff_isNontrivial.mpr ‹_›
+
+end Valuation
+
+namespace Polynomial
+
+open WithZero WithZeroMulInt NNReal
+
+variable (K : Type*) [Field K]
+
+/-- The valuation on `K[X]` that corresponds to the "point at infinity" on the projective line `ℙ¹`,
+sending `X` to `WithZero.exp 1`. -/
+noncomputable def infinityValuation : Valuation K[X] ℤᵐ⁰ :=
+  ((idealX K).valuation (RatFunc K)).comap <| RingHomClass.toRingHom <| aeval .X⁻¹
+
+@[simp] lemma infinityValuation_X : infinityValuation K X = exp 1 := by
+  simp [infinityValuation, exp]
```

Comment:
this is the inverse of the "degree" valuation, correct?

Reply:
I think it's exactly the degree, no inverse needed; do we have that in mathlib?

Reply:
No, it's still not inversed. You don't get a valuation by applying inverse, because the X-adic valuation of (1+X) is 0, and the degree valuation of (1+X) is 1.

Reply:
The X-adic valuation is the inverse of the trailing degree, and there is no linkage between those two and the degree valuation.